### DISCUSSION TITLE
Clarification on dqlite read operations

### DISCUSSION THREAD

**Discussion by Roberto-Mameli** [canonical/dqlite#599] [Q&A]
Hi all,

from dqlite documentation I understand that in case of replication, there is a node that is elected as leader in the cluster. This node manages SQL statements and provides back answers to the client.
There is a detail that is not clear to me. Does the mechanism described above applies both to SQL write and SQL read statements? My first understanding was that SQL write statements shall be always forwarded to the cluster leader, but the SQL read statements (i.e. SELECT) can be managed by any replica. However, I do not find any explicit indication of that, so I'm not sure that it corresponds to the actual behaviour.
Can you please provide a clarification about the behaviour of read statements (and possibly specify it explicitly in the documentation as well)?

Thank you in advance
Roberto

**Comment by cole-miller** [comment]
> My first understanding was that SQL write statements shall be always forwarded to the cluster leader, but the SQL read statements (i.e. SELECT) can be managed by any replica. However, I do not find any explicit indication of that, so I'm not sure that it corresponds to the actual behaviour.

Read requests are also forwarded to the leader in the current design, although this is something we hope to change in the future as it's a common concern/pain point.

**Comment by computinglife** [comment]
Hi, 

Are the reads still directed to the leader? 

Thanks in advance
Arun
